### PR TITLE
Added repository user authorization support + server flag refactoring + refresh

### PR DESCRIPTION
**Pull request by jkowalski**

### CODE REVIEW

File: cli/sighup_unix.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+)
+
+// onExternalConfigReloadRequest invokes the provided function when SIGHUP is received.
+func onExternalConfigReloadRequest(f func()) {
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, syscall.SIGHUP)
+
+	go func() {
+		<-c
+		f()
```
Comment (julio-lopez) [inline cli/sighup_unix.go:18]: Does this need to be in a loop?

**Review by julio-lopez — COMMENTED** [review]
👍 LG +1

File: cli/sighup_unix.go (hunk 1 above)
Comment (jkowalski) [reply cli/sighup_unix.go]: done

**Review by julio-lopez — COMMENTED** [review]
👍 LG +1

**Pull request merged**